### PR TITLE
*Support wildcard and subdomains

### PR DESCRIPTION
**Pull request by pcfreak30**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -80,7 +78,9 @@ server.get("/dnslink/:name", async (req, res) => {
     console.log(`${req.params.name} => ${skylink}`);
 
     return success({ skylink });
-  });
+  }
+
+  return failure(`No TXT record found for _dnslink.${req.params.name}`);
```
Comment (interstella-5555) [inline src/index.js:83]: There is a number of specific user facing errors that have been squashed as this generic one, degrading logs potential debugging usability and less informative for end user that tries to set up dnslink on a domain.

**Review by interstella-5555 — COMMENTED** [review]
@pcfreak30 can you write down what is your use case for this feature? I think I have an idea how to improve this solution but I would like to ensure I know what is your specific use case first.

cc @dghelm

**Comment by pcfreak30** [comment]
@kwypchlo This is to enable for example ans.algo.website, https://twitter.com/algonameservice/status/1529896088689180672. It is a single dapp that has a wildcard CNAME to fileportal, but this enables it to detect the domain on load and show a social page.

additionally, it is enabling www.web3extension.com as well as web3extension.com (a lume web website) with a single TXT record with the same logic. The fact it is generic means it solves a lot of situations in one go.

I had explained this before @MSevey rebuilt the repo :sweat_smile: and my pr was deleted.

To be clear this is currently in production and is working as intended for my and ANS's usecases.

**Comment by interstella-5555** [comment]
> @kwypchlo This is to enable for example ans.algo.website, https://twitter.com/algonameservice/status/1529896088689180672. It is a single dapp that has a wildcard CNAME to fileportal, but this enables it to detect the domain on load and show a social page.
> 
> additionally, it is enabling [www.web3extension.com](http://www.web3extension.com) as well as web3extension.com (a lume web website) with a single TXT record with the same logic. The fact it is generic means it solves a lot of situations in one go.
> 
> I had explained this before @MSevey rebuilt the repo 😅 and my pr was deleted.
> 
> To be clear this is currently in production and is working as intended for my and ANS's usecases.

Sorry I couldn't get to that earlier. I will talk with @dghelm today about that.

**Comment by dghelm** [comment]
Let's talk more Karol. But, from a brief glance, I don't find these use cases very compelling for deviating from how DNSLink works as a standard. I think both of these use cases can even be done using features like [Cloudflare's Wildcard proxy](https://blog.cloudflare.com/wildcard-proxy-for-everyone/).

The main issue that bothers me is that, as far as I can tell, this would encourage webapps to design themselves in such a way as to break their compatibility with access-via-skylink. This makes it very difficult to access a webapp through any means outside of their chosen domain-name system, and constrains users.

For example: algo.website uses DNSLink to point to `sia://AQCQ8ah-tqWW6sYQcguTFPpeCfop7Z16ih0tuL_eDUC0lw` but because of how the routing logic for detecting the subdomains works, their own homepage is [inaccessible from a portal url](https://04091sd8fqrab5naoo8742sj2jt5s2fq57mpquka3kmrhfuu1l0b95o.siasky.net/), and another layer of subdomains for user accounts further breaks skylink paradigms.

**Comment by pcfreak30** [comment]
```For example: algo.website uses DNSLink to point to sia://AQCQ8ah-tqWW6sYQcguTFPpeCfop7Z16ih0tuL_eDUC0lw but because of how the routing logic for detecting the subdomains works, their own homepage is [inaccessible from a portal url](https://04091sd8fqrab5naoo8742sj2jt5s2fq57mpquka3kmrhfuu1l0b95o.siasky.net/), and another layer of subdomains for user accounts further breaks skylink paradigms.```

It was never meant to be accessible from a portal URL because it would not work that way even if you could fix that. Also portal URLs long-term should not matter anyways unless you're using it as a basic filestore or managing your account. Regardless things are working as designed and are serving their purpose. Also to restate web3extension.com does the same with `www.` which simplifies it all.

@dghelm @kwypchlo

**Comment by pcfreak30** [comment]
`I think both of these use cases can even be done using features like [Cloudflare's Wildcard proxy](https://blog.cloudflare.com/wildcard-proxy-for-everyone/).` I am not sure this is the case as you need it for the TXT record, not the CNAME record and I already experimented.

**Pull request closed without merging**